### PR TITLE
Add request for non-local variable definition

### PR DESCRIPTION
**Pull request by chrismwendt**
Part of https://github.com/sourcegraph/sourcegraph/issues/35031

Depends on https://github.com/sourcegraph/sourcegraph/pull/35082

This requests a non-local variable definition if a local one could not be found.

**Reviewers: ignore the GraphQL changes - they're synced from sourcegraph/sourcegraph**

### CODE REVIEW

File: schema/schema.graphql [hunk 1]
```diff
@@ -2290,6 +2291,7 @@ enum ExternalServiceKind {
     PAGURE
     PERFORCE
     PHABRICATOR
+    PYTHONPACKAGES
```
Comment (varungandhi-src) [inline schema/schema.graphql:2294]: Q: How is this change + the change adding GERRIT above related to the rest of this patch? Did those changes sneak in through rebasing?

File: schema/schema.graphql [hunk 2]
```diff
@@ -3768,6 +3770,15 @@ type GitCommit implements Node {
...
     """
+    The Git tree or blob in this commit at the given path.
+    """
+    path(
+        """
+        The path of the tree or blob.
+        """
+        path: String = ""
+    ): GitTreeOrBlob
+    """
```
Comment (varungandhi-src) [inline schema/schema.graphql:3781]: [Warning: GraphQL noob question] I'm a little confused... There is already a `GitTree` below and a `GitBlob` as well. Could you clarify why this field is needed?

File: schema/schema.graphql (hunk 1 above)
Comment (chrismwendt) [reply schema/schema.graphql:2294]: This GraphQL file hasn't been updated in a while, so syncing it pulled in a lot of unrelated changes.

File: schema/schema.graphql (hunk 2 above)
Comment (chrismwendt) [reply schema/schema.graphql:3781]: (this is an unrelated change that got synced)

`GitTreeOrBlob` appears to be an alias for `GitTree | GitBlob`.

**Review by cesrjimenez — APPROVED** [review]
LGTM! 🚀

**Pull request merged**